src/server/project.ts

```diff
@@ -457,6 +457,11 @@ namespace ts.server {
             return this.getTypeAcquisition().enable ? this.projectService.typingsInstaller.globalTypingsCacheLocation : undefined;
         }
 
+        /*@internal*/
+        fileIsOpen(filePath: Path) {
+            return this.projectService.openFiles.has(filePath);
+        }
+

```

Comment:
You want to also check this in `watchWildCardDirectory` in the `editorServices.ts`

Reply:
Also need test for that scenario since the log you showed me was hitting this scenario.

Reply:
I'm happy to do that, but the case I was specifically trying to avoid was the resolution cache invalidation, which doesn't happen in the wild card directory case (as far as I can tell). If you think it's appropriate to not trigger an `updateGraph` in the wild card directory scenario at all, then I can definitely make the change, but I just wanted to make sure since that's a more substantial change than I was planning on making.

Reply:
Yes we should definitely do that in the watchwildcard directory as in your repro that’s what caused the issue

Reply:
I've added the wild card scenario. I don't necessarily agree that the wild card directory watcher is what caused the issue in my repro; rather, the issue occurs when at least one of the directory watchers triggered from a file save operation is a failed lookup watcher, and it just so happens that it usually (always?) happens in conjunction with a wild card watcher being triggered.  However, I agree that it seems fine, even beneficial, to not schedule a graph update when the wild card watcher is triggered by an open file being saved, so I've added that.